src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -121,10 +181,46 @@ public void UpdateWorkloads(bool includePreviews = false, DirectoryPath? offline
             Reporter.WriteLine();
         }
 
+        private IEnumerable<ManifestVersionUpdate> CalculateManifestUpdates(WorkloadHistoryRecorder recorder)
+        {
+            if (!string.IsNullOrWhiteSpace(_fromHistorySpecified))
+            {
+                WorkloadHistoryState state = _WorkloadHistoryRecord.StateAfterCommand;
+
+                var currentWorkloadState = WorkloadRollbackInfo.FromManifests(_workloadResolver.GetInstalledManifests());
+
+                var versionUpdates = new List<ManifestVersionUpdate>();
+
+                foreach (KeyValuePair<string, string> m in state.ManifestVersions)
+                {
+                    var manifestId = new ManifestId(m.Key);
+                    var featureBandAndVersion = m.Value.Split('/');
+                    var currentVersionInformation = _workloadManifestUpdater.GetInstalledManifestVersion(manifestId, throwIfNotFound: false);
+                    var band = currentVersionInformation?.Band;
+                    versionUpdates.Add(new ManifestVersionUpdate(
+                        manifestId,
+                        currentVersionInformation?.Version,
+                        band?.ToString(),
+                        new ManifestVersion(featureBandAndVersion[0]),
+                        featureBandAndVersion[1]));
+                }
```

Comment:
This seems similar to what the manifest updater is doing in `CalculateManifestRollbacks` and `CalculateManifestUpdates`.  So perhaps the logic should be moved there, and possibly refactored to reduce duplication.

Reply:
They have some pieces in common, most notably the structure of the final output and the general goal...looking at it now, though, I think they have too many subtle differences for code sharing to be practical. I can certainly move it over, though.